types/node-horseman/index.d.ts

```diff
@@ -0,0 +1,250 @@
+// Type definitions for node-horseman 3.3
+// Project: https://github.com/johntitus/node-horseman#readme
+// Definitions by: Pratheek Adidela <https://github.com/apratheek>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 2.2
+
+/*~ If this module is a UMD module that exposes a global variable 'myLib' when
+ *~ loaded outside a module loader environment, declare that global here.
+ *~ Otherwise, delete this declaration.
+ */
+
+export = horseman;
+
+declare namespace horseman {
+    class horseman {}
```

Comment:
do not see this in the node package:

```sh
> npm install  node-horseman

+ node-horseman@3.3.0
added 9 packages in 1.963s

> node
> var n = require("node-horseman");
undefined
> n;
{ [Function: Horseman]
  registerAction: [Function: registerAction],
  TimeoutError: [Function: SubError] }
> n.horseman
undefined

```

Reply:
Hi, 

The package is used as follows:
```
const horseman = require('node-horseman');
let horse = new horseman({
    timeout: 10000
});
horse.on('error', (msg: any, trace: any) => { console.log(msg, trace); })
+.open("http://example.org/").waitForNextPage().open("https://google.com").waitForNextPage().html().then((pageContent: string) => {
+    console.log(pageContent);
+    horse.close();
+});
```

The package requires `PhantomJS` (npm install phantomjs). I think I may not have added this as a dependency. Is that the issue? This is my first package, so I'm out of my depth here. Any help would be appreciated.

Reply:
I do not think you need this namespace/inner class. your other class declaration for `horseman` should be sufficient.

Reply:
Hi, I tried to remove the namespace and directly export the class. It kinda made the initialization complicated, plus, I couldn't really get it to run on Node. Using the current structure, it works, though. Just to let you know, I'm an amateur at Typescript, so I don't exactly know how classes and namespaces go together.

I tried to run the test on my end, and it works. Am I missing something here?

Reply:
Just delete the 3 lines of `declare namespace horseman` entirely and the code should work assuming you change the import style as shown in my other comment.